ci/pipeline

```diff
@@ -273,9 +285,79 @@ def release(requestVersion: String, gitSha: String, latest: Boolean = false): Un
   // TODO: Publish native packages to unstable.
 }
 
+/**
+  * Given tagName v1.5.1, and rev DEADBEEF, we would assert that v1.5.0 is a direct ancestor of rev DEADBEEF
+  *
+  * Also, we assert that rev contains new commits since the last release. If the proposal is to tag v1.5.1 with same
+  * revision as v1.5.0, then we fail, saying that there v1.5.0 points to the commit porposed for v1.5.1
+  *
+  * @param tagName The tag name / version we are going to tag. eg v1.5.1
+  * @param gitSha The commit sha / rev that will be tagged.
+  */
+def assertPriorVersionIsAncestor(tagName: String, gitSha: String): Unit = {
+  utils.priorPatchVersion(tagName).foreach { prior =>
+    val revs = %%("git", "rev-list", "--format=oneline", s"${gitSha}..${prior}").out.string
+    if(revs.trim.nonEmpty) {
+      println(s"Error! ${prior} is not a direct ancestor of proposed commit ${gitSha} for ${tagName}!")
+      sys.exit(1)
+    }
+
+    val commitsSinceLast = %%("git", "rev-list", "--format=oneline", s"${prior}..${gitSha}").out.string
+    if(commitsSinceLast.trim.isEmpty) {
+      println(s"Error! ${prior} points to proposed commit ${gitSha} for ${tagName}!")
+      sys.exit(1)
+    }
+  }
+}
+
 def withoutVersionPrefix(releaseVersion: String): String = {
     if(releaseVersion.toLowerCase.startsWith("v"))
       releaseVersion.substring(1)
     else
       releaseVersion
 }
+
+def uploadLinuxPackagesToRepos(tagName: String): Unit = {
+  val pkgserverUser = sys.env.getOrElse("PKG_SSH_USER", {
+    throw new IllegalStateException("PKG_SSH_USER environment variable must be set")
+  })
+  val pkgserverHost = sys.env.getOrElse("PKG_SSH_HOST", {
+    throw new IllegalStateException("PKG_SSH_HOST environment variable must be set")
+  })
+
+  // Note - the key is expected to be provided via an SSH agent
+  utils.printStageTitle(s"Uploading native packages")
+  %("rsync", "-avz",
+    (pwd / 'target / 'packages) + "/",
+    s"${pkgserverUser}@${pkgserverHost}:repo/incoming/marathon-${tagName}/")
```

Comment:
So we first rsync the packages and then move them on the server? Where is the unstable channel defined?

Reply:
nothing yet created for the unstable channel.     Perhaps we will have agreement out of our release meeting.